sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala

```diff
@@ -846,18 +859,14 @@ object CreateNamedStructCheck extends ExprChecks {
     if (context != ProjectExprContext) {
       meta.willNotWorkOnGpu(s"this is not supported in the $context context")
     } else {
-      val origExpr = exprMeta.wrapped.asInstanceOf[Expression]
-      val (nameExprs, valExprs) = origExpr.children.grouped(2).map {
-        case Seq(name, value) => (name, value)
-      }.toList.unzip
-      nameExprs.foreach { expr =>
-        nameSig.tagExprParam(meta, expr, "name")
-      }
-      valExprs.foreach { expr =>
-        valueSig.tagExprParam(meta, expr, "value")
+      exprMeta.childExprs.grouped(2).foreach {
+        case Seq(nameMeta, valueMeta) =>
+          nameSig.tagExprParam(meta, nameMeta, "name")
+          nameSig.tagExprParam(meta, valueMeta, "value")
       }
-      if (!resultSig.isSupportedByPlugin(origExpr.dataType, meta.conf.decimalTypeEnabled)) {
-        meta.willNotWorkOnGpu(s"unsupported data type in output: ${origExpr.dataType}")
+      val dt = exprMeta.dataType.get
+      if (!resultSig.isSupportedByPlugin(dt, meta.conf.decimalTypeEnabled)) {
+        meta.willNotWorkOnGpu(s"unsupported data type in output: $dt")
       }
```

Comment:
nit
```suggestion
      exprMeta.dataType.foreach { dt => 
        if (!resultSig.isSupportedByPlugin(dt, meta.conf.decimalTypeEnabled)) {
          meta.willNotWorkOnGpu(s"unsupported data type in output: $dt")
        }
      }
```

Reply:
Done. Thanks!

Reply:
Did you forget to push, I actually don't see this reflected in the PR

Reply:
Yes, I forgot it. And it's done now.